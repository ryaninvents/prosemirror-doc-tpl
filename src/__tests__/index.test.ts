import createBuilder from '../index';
import { Fragment } from 'prosemirror-model';
import { schema } from 'prosemirror-schema-basic';
import * as Util from 'prosemirror-test-builder';

const builder = Util.builders(schema);
const pm = createBuilder(schema);

describe('createBuilder', () => {
  it('should create a basic document', () => {
    const document = pm`
          <doc>
            <heading level=1>Hello world</heading>
            <heading level=2>And now for something completely different</heading>
            <paragraph>
              Here's to the crazy ones. The misfits. The rebels...
            </paragraph>
          </doc>
        `;
    expect((document as any).toJSON()).toMatchSnapshot();
  });
  it('should properly handle bare marks', () => {
    const frag = pm`<>Here is some <strong>bold text</strong></>`;
    const targetJson = [
      schema.text('Here is some ').toJSON(),
      ...(Fragment.from(
        builder.strong({}, schema.text('bold text')).flat
      ).toJSON() as any[]),
    ];
    expect(Fragment.from(frag).toJSON()).toEqual(targetJson);
  });
  it('should properly nest marks', () => {
    const doc1 = pm`
          <doc>
            <paragraph>
              Here is some <strong>bold text</strong>
            </paragraph>
          </doc>
        `;
    expect((doc1 as any).toJSON()).toMatchInlineSnapshot(`
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Here is some ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": "bold text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`);
  });
});
