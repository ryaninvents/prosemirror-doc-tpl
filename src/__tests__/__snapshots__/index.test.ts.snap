// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBuilder should create a basic document 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": "1",
      },
      "content": Array [
        Object {
          "text": "Hello world",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": "2",
      },
      "content": Array [
        Object {
          "text": "And now for something completely different",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "content": Array [
        Object {
          "text": "Here's to the crazy ones. The misfits. The rebels...",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
